### ISSUE TITLE
VP3 color blocks slightly offset

### ISSUE THREAD

**Issue by lexelby** [inkstitch/inkstitch#205]
When saving as VP3 and loading it back up with Ink/Stitch, color blocks seem to be slightly offset:

![vp3_broken](https://user-images.githubusercontent.com/4446653/41494665-78f85126-70e4-11e8-8278-aa090cd119fa.png)

Otherwise, it looks great.

**Comment by tatarize** [comment]
To test them, you should not only export and import and make sure those fit, but try something like loading them in Wilcom TruSizer or OESD .artSizer, there's a few more errors you likely didn't find, on top of that they also have to be checked for consistency by actually sewing them on a machine. There's also sometimes also errors that only crop up in one place like load .PES files in brother's software vs. something that only reads the stitches like Wilcom.

LibEmbroidery exports .vp3 to give a manufacturer string in utf-8 while Wilcom writes that string in utf-16 and some of the thread strings in utf-8. I can't imagine nothing cares if that's what Wilcom is exporting. It's the consistency and format part that is the bulk of this work and why `pyembroidery` came together so quickly. I had already done loads of that work in the Java version and have enough people actually using the sewouts to get some feedback. And looked down at the same general problems that the libembroidery code just wasn't consistent enough. And to get consistent enough, people need to do things like actually figure out what all the parts of the format actually do, like was done for .PES (save some of the mysterious parts in the PEC blocks).

I'm currently working on the vp3 stuff using embroidermodder c++ code as a basis, and it's taking hours longer than most of the other readers and writers and still doesn't work at all.  You won't have found all the errors. I know from previously fixing the embroidermodder code in java, there's a couple other landmines out there. But, if you check such things and figure out that I'm right about such things, it'll accelerate your adoption of pyembroidery, and hopefully start moving any issues with the reading and writing over to that project, just as properly these broken file formats issues should be handed off to libembroidery, because other than know about them, you can't actually fix them. Also, those jump stitches turning into stitches stitches is an issue too.

**Comment by tatarize** [comment]
Pretty sure I found this bug. The colorblock initial stitch location for the first block is not relative to the actual position of the thread but rather 0,0 relative to the design center. It doesn't do much, but causes the setting of the absolute position within first color block to be off by the distance of the first stitch. 

It might also be rounding error of the recorded centerx, centery value not being exactly centerx, centery but all calculations must be relative to that position so it must be maintained properly rather than writing it down and not fixing the values relative to that rounding error (which would only crop up in the layers themselves).

In any event, I firmly believe this is fixed in pyembroidery (should be, I mapped out the entire format and corrected all the unknowns, to the point of being able to free read it in binary). Wilcom seems to screw the first one there up, I think embroidermodder is screwing up the second potential pitfall there.
